### DISCUSSION TITLE
Is incremental adoption of TypeGraphQL possible?

### DISCUSSION THREAD

**Discussion by tony-oldport** [MichalLytek/type-graphql#1374] [Q&A] [answered]
I would like to incrementally migrate my project to TypeGraphQL, but the issue I run into is when I'm defining Fields that are of a type that hasn't been declared yet as an ObjectType, I get the error: 

`Unable to infer GraphQL type from TypeScript reflection system. You need to provide explicit type for 'x' of 'y' class`

In other words, if a type only exists in my manually declared GraphQL schema and has not yet been migrated to an TypeGraphQL ObjectType, I can't use it as a Field type. Defining ObjectTypes as I need them will basically just cascade into migrating the entire project.

Is there some generic type I could add to the Field decorator to make this work? Or maybe there's a simpler solution that I haven't managed to find in other discussions and examples.

`@Field()`
` user: any`

**Comment by tony-oldport** [comment]
I think I may be asking a very similar question to this, just in a slightly different way: https://github.com/MichalLytek/type-graphql/discussions/1176

**Comment by MichalLytek** [comment, marked as answer]
No, you can't refer somewhere existing type name. However you can try to migrate types first, leaving resolvers in the old way and merge the declarations.
There are also tools/plugins to generate TypeGraphQL types from existing schema 😀

**Reply by tony-oldport** [reply]
Great, thanks. Migrating types first seems pretty reasonable.
